jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3352,9 +3352,16 @@ public String[] split(String regex, int max) {
 			} else {
 				int rLength = regex.lengthInternal();
 
-				byte[] splitChars = regex.value;
+				byte[] splitChars;
+				// if regex is compressed and this string isn't, decompress regex string
+				if (COMPACT_STRINGS && !compressed && regex.isCompressed()) {
```

Comment:
Testing `COMPACT_STRINGS` is redundant. I'd like an opinion from the JIT team whether it's worth keeping that term.

Reply:
@dchopra001 Do you have any thoughts about this?

Reply:
@keithc-ca From the JIT's perspective we shouldn't restrict any optimization by removing `COMPACT_STRINGS` here.

Reply:
Removed the check for `COMPACT_STRINGS` in https://github.com/eclipse-openj9/openj9/commit/7251c94d5fc4056e8389573efd1084b674153905